lib/spack/spack/ci.py

```diff
@@ -1343,13 +1343,9 @@ def generate_gitlab_ci_yaml(
         sorted_output = {"no-specs-to-rebuild": noop_job}
 
     if known_broken_specs_encountered:
-        error_msg = (
-            "Pipeline generation failed due to the presence of the "
-            "following specs that are known to be broken in develop:\n"
-        )
-        for broken_spec in known_broken_specs_encountered:
-            error_msg += "* {0}\n".format(broken_spec)
-        tty.die(error_msg)
+        tty.error("This pipeline generated hashes known to be broken on develop:")
+        display_broken_spec_messages(broken_specs_url, known_broken_specs_encountered)
+        tty.die()
```

Comment:
Yeah, sorry.  My test didn't notice it because it has to pass `allow_failure=True` to the SpackCommand, since the command is expected to fail in this case and the expected text was in the output.  I added a fix [here](https://github.com/spack/spack/pull/32596/commits/bc56abc68e355e3d885574cb03bd862733bc8ebb) since I was ready to merge that, but have had to retry the py27 and py38 tests more times on that PR.  If that requires any more retries, I'll make a PR just for this.